web-ui/src/main/resources/catalog/components/common/map/mapService.js

```diff
@@ -1432,19 +1442,41 @@
             return defer.promise;
           },
 
+          /**
+           * @ngdoc method
+           * @methodOf gn_map.addEsriRestFromScratch:gnMap
+           * @name gnMap#addEsriRestFromScratch
+           *
+           * @description
+           * Here is the method to use when you want to add a ESRIREst layer from
+           * a url and a name (layer identifier).
+           *
+           * Return a promise with ol.Layer as data is succeed, and url/name
+           * if failure.
+           * If createOnly, we don't add the layer to the map.
+           * If the md object is given, we add it to the layer, or we try
+           * to retrieve it in the catalog
```

Comment:
This contradicts the name of the methods wich ends with `FromScratch`; I think it makes sense to rename this method `addEsriRestLayer` for example

Reply:
This method already existed, it's not from this pull request. Can you explain why contradicts the name of the methods which ends with `FromScratch`?

Reply:
> This method already existed, it's not from this pull request.

I know, but this doesn't mean we cannot rename it.

"From scratch" means, in the context of the GN map service, that it does not need to be fed a capabilities document; besides, there is usually another "from capabilities" function that comes with it. For ESRI services it does not work like that so I suggest we remove the "from scratch" from the method name as it is misleading.

Reply:
Ok, I don't see that much of an issue to use the `FromScratch` suffix (the method was already there, hasn't been added in this pull request), as the method doesn't use a capabilities document. But if it makes, the code more clear will update it.

Reply:
I understand, but I also disagree: while it may appear minor, an incorrect/misleading function name can significantly diminish the readability of the code. A widely accepted good practice is that whichever piece of code you alter should be cleaner after your changes than prior; this can include minor refactorings like this.

That said, I was simply making a comment about something that striked me as incorrect, not specifically requesting changes on your behalf.

Reply:
It's already updated